### PR TITLE
Review docs

### PR DESCRIPTION
**Pull request by UZerbinati**
Doc review in view of JOSS submission.

### CODE REVIEW

File: ngsPETSc/ts.py [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+    def __init__(self, fes, info, G=None, F=None, residual=None,
+                 jacobian=None, solverParameters=None, optionsPrefix=None):
+        self.fes = fes
+        dofs = fes.ParallelDofs()
+        self.second_order = False
+        self.trial = info[0]
+        self.t = info[1]
+        self.F = F
+        self.G = G
+        if "ngs_jacobian_mat_type" not in solverParameters:
```
Comment (stefanozampini) [inline ngsPETSc/ts.py:25]: if you do this, you need to use `solverParameters={}` not `solverParameters=None`

File: ngsPETSc/ts.py [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+            solverParameters["ngs_jacobian_mat_type"] = "aij"
+        jacobianMatType = solverParameters["ngs_jacobian_mat_type"]
+        self.ts = PETSc.TS().create(comm=dofs.comm.mpi4py)
+        #Deafult TS setup
+        self.ts.setExactFinalTime(3)
```
Comment (stefanozampini) [inline ngsPETSc/ts.py:30]: use class enums, not integers https://petsc.org/release/petsc4py/reference/petsc4py.PETSc.TS.ExactFinalTime.html#petsc4py.PETSc.TS.ExactFinalTime

File: ngsPETSc/ts.py [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+        This is method is used to setup the PETSc TS object
+        '''
+        if isinstance(timeInfo, (list,tuple)) and len(timeInfo) == 3:
+            self.t0, self.tf, self.dt = timeInfo
+        elif isinstance(timeInfo, dict):
+            self.t0 = timeInfo["t0"]
+            self.tf = timeInfo["tf"]
+            self.dt = timeInfo["dt"]
+        else:
+            raise ValueError("timeInfo must be a list/tuple or a dict")
```
Comment (stefanozampini) [inline ngsPETSc/ts.py:90]: why timeInfo is here ?
these should be quantities decided at initialization time, not setUp.
Take a look at how I do TS customization in deal.II https://github.com/dealii/dealii/blob/master/include/deal.II/lac/petsc_ts.h#L44

File: ngsPETSc/ts.py [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+                a += self.G
+                a.Apply(x.vec, f.vec)
+                return f
+            self.rhs = rhs
+        if F is not None:
+            def residual(t, x, xdot): #pylint: disable=E0102, E0213
+                res = GridFunction(self.fes)
+                self.t.Set(t)
+                updatedF = self.F.Replace({self.trial.dt: xdot})
+                a = BilinearForm(self.fes)
```
Comment (stefanozampini) [inline ngsPETSc/ts.py:55]: Take a look here https://github.com/dealii/dealii/blob/master/include/deal.II/lac/petsc_ts.templates.h#L432

**Review by stefanozampini — COMMENTED** [review]
Only TS so far. The code is a little convoluted, are you sure you want to push it to main?

**Comment by UZerbinati** [comment]
Docs and JOSS paper.

**Pull request merged**